action.yml

```diff
@@ -339,16 +339,21 @@ runs:
           echo "release=${DISTRIB_RELEASE}" >> $GITHUB_OUTPUT
           echo "codename=${DISTRIB_CODENAME}" >> $GITHUB_OUTPUT
           echo "description=${DISTRIB_DESCRIPTION}" >> $GITHUB_OUTPUT
-          echo "id-release=${DISTRIB_ID}-${DISTRIB_DESCRIPTION}" >> $GITHUB_OUTPUT
+          echo "id-release=${DISTRIB_ID}-${DISTRIB_RELEASE}" >> $GITHUB_OUTPUT
           echo "arch=$(uname -m)" >> $GITHUB_OUTPUT
         elif [ "$RUNNER_OS" == "macOS" ]; then
           echo "id-release=macOS-$(sw_vers -productVersion)" >> $GITHUB_OUTPUT
           echo "arch=$(uname -m)" >> $GITHUB_OUTPUT
         fi
+        # Create an envvar that stores information about the action version used
+        ACTION_VERSION=$(basename "$GITHUB_ACTION_PATH")
+        ACTION_NAME=$(basename $(dirname "$GITHUB_ACTION_PATH"))
+        ACTION_OWNER=$(basename $(dirname $(dirname "$GITHUB_ACTION_PATH")))
+        echo "cvmfs_action_version=$ACTION_OWNER_$ACTION_NAME_$ACTION_VERSION" >> $GITHUB_OUTPUT
       shell: bash
     - uses: actions/cache@v4
       with:
-        key: cvmfs-apt-cache-${{ steps.lsb-release.outputs.id-release }}-${{ steps.lsb-release.outputs.arch }}-${{ hashFiles('action.yml') }}
+        key: cvmfs-apt-cache-${{ steps.lsb-release.outputs.id-release }}-${{ steps.lsb-release.outputs.arch }}-${{ steps.lsb-release.outputs.cvmfs_action_version }}
```

Comment:
@vvolkl I think this will work but it is tricky to test properly, the cache in this case is called
```
cvmfs-apt-cache-Ubuntu-22.04-x86_64-.
```
because the actions is `./`

Reply:
I'll see if I can trigger it in my own project CI

Reply:
Ok, that is fixed now and I see
```
cvmfs-apt-cache-Ubuntu-24.04-aarch64-cvmfs-contrib_github-action-cvmfs_987d52a1a6c90e0a3304453c3f606fa45f376bd0
```

Reply:
> Ok, that is fixed now and I see
> ```
> cvmfs-apt-cache-Ubuntu-24.04-aarch64-cvmfs-contrib_github-action-cvmfs_987d52a1a6c90e0a3304453c3f606fa45f376bd0
> ```

Is the originally highlighted issue that there might be a length limit on cache keys mitigated somehow, or not a concern anymore?

Reply:
That was a red herring from me, I don't see any issues with that in my latest tests

Reply:
It really seemed to be related to hashing a file (`action.yml`) that didn't exist in the repo using the Action, leading to cache misses

Reply:
Hmm, maybe there is an easier solution, the file _should_ exist under `$GITHUB_ACTION_PATH`

Reply:
Reading https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#hashfiles , I don't think you can use `hashfiles` with `$GITHUB_ACTION_PATH/action.yml`

Reply:
> In that case, there are some reports about the github.action_path handlebar variable (which I assume is the same as the env var) pointing to non-existing files in self-hosted runners. [actions/runner#716](https://github.com/actions/runner/issues/716)
> 
> Not sure if that's of potential concern. I don't even know if this action works on self-hosted runners right now.

The workaround there is to use the environment variable instead, and the current approach (accidentally) already does that